### PR TITLE
Fixes #7561: Use Scandir to speed up our os.walk usage

### LINKED ISSUE #7561: Use Scandir to speed up our os.walk usage
One of our dev dependancies uses scandir and the description implies it's double++good
https://github.com/mozilla/addons-server/pull/7549
We should use it in prod code too (?)

### PR DESCRIPTION
**Pull request by CuriousLearner**
Used scandir instead of `os.walk` to speed up usage of `walk` method.
I've imported it as `scandir_walk` to explicitly keep name-spacing consistent. This would help people in future to quickly know that it is not normal walk that is shipped with `os` module.

Please delete anything that isn't relevant to your patch.

* [x] This PR relates to an existing open issue and there are no existing
      PRs open for the same issue.
* [x] Add `Fixes #ISSUENUM` at the top of your PR.
* [x] Add a description of the the changes introduced in this PR.

Let me know if this needs any further changes ;)

### CODE REVIEW

File: src/olympia/amo/utils.py [hunk 1]
```diff
@@ -40,6 +40,7 @@
 import pytz
 
 from babel import Locale
+from scandir import walk as scandir_walk
```
Comment (eviljeff) [inline src/olympia/amo/utils.py]: just do `import scandir` so you can replace `os...` with `scandir...` all over.

**Comment by CuriousLearner** [comment]
@eviljeff Done :)

**Comment by CuriousLearner** [comment]
@eviljeff Not sure why the circle CI is failing on master and showing that `scandir` wasn't found and could not be imported. I see that it is already part of requirements in `dev.txt`. Can you please look into this?

**Comment by eviljeff** [comment]
Ah! Good call, the requirement should in prod.txt instead.

**Comment by CuriousLearner** [comment]
Ah, sorry, I didn't know that. I'm sending a patch right away :)

**Comment by eviljeff** [comment]
Thanks!  (I didn't remember either - and Travis passing fooled me)

**Comment by CuriousLearner** [comment]
@eviljeff Here is the new PR: https://github.com/mozilla/addons-server/pull/7730

**Pull request merged**